lib/api/controller/auth.js

```diff
@@ -326,7 +387,14 @@ class AuthController extends NativeController {
    * @returns {Promise<object>}
    */
   async checkToken (request) {
-    const token = this.getBodyString(request, 'token');
+    let token = '';
+
+    if (!global.kuzzle.config.security.supportCookieAuthentication || !request.input.args.cookieOnly) {
+      token = this.getBodyString(request, 'token');
+    }
+    else {
+      token = request.input.jwt;
+    }
```

Comment:
Nitpicking: (code complexity) it's more readable to have a logical AND rather than NOR ... OR NOR ...

```suggestion
    if ( global.kuzzle.config.security.supportCookieAuthentication 
      && request.input.args.cookieOnly
    ) {
      token = request.input.jwt;
    }
    else {
      token = this.getBodyString(request, 'token');
    }
```